src/blocked.c

```diff
@@ -231,15 +229,19 @@ void replyToBlockedClientTimedOut(client *c) {
         addReplyNullArray(c);
         updateStatsOnUnblock(c, 0, 0, 0);
     } else if (c->bstate.btype == BLOCKED_WAIT) {
-        addReplyLongLong(c, replicationCountAcksByOffset(c->bstate.reploffset));
-    } else if (c->bstate.btype == BLOCKED_WAITAOF) {
-        addReplyArrayLen(c, 2);
-        addReplyLongLong(c, server.fsynced_reploff >= c->bstate.reploffset);
-        addReplyLongLong(c, replicationCountAOFAcksByOffset(c->bstate.reploffset));
+        if (c->cmd == shared.wait_cmd) {
+            addReplyLongLong(c, replicationCountAcksByOffset(c->bstate.reploffset));
+        } else if (c->cmd == shared.waitaof_cmd) {
+            addReplyArrayLen(c, 2);
+            addReplyLongLong(c, server.fsynced_reploff >= c->bstate.reploffset);
+            addReplyLongLong(c, replicationCountAOFAcksByOffset(c->bstate.reploffset));
+        } else if (c->cmd == shared.setslot_cmd) {
```

Comment:
CLUSTER SETSLOT is handled by clusterCommand, just like all cluster subcommands. We can either break it out to its own function clusterSetslotCommand and update the JSON file, or we can just check for clusterCommand here. It's the only cluster command that does wait anyway. (It that ever changes, we can handle refactor then.)

```suggestion
        } else if (c->cmd->proc == clusterCommand) {
```